sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/google-oauth2.yaml

```
@@ -0,0 +1,227 @@
+          containsSecret: true
+          match: TgxYWFmND-1NTYwNTgzMDM3N
+        - text: >
+            self.google_app = SocialApp.objects.create(
+              provider="google",
+              name="Google OAuth",
+              client_id="514233951460-ptar9o2diyaiz06uxa9inr2iuqjlvg1u.apps.googleusercontent.com",
+              secret="TgxYWFmND-1NTYwNTgzMDM3N",
+            )
+          containsSecret: true
+          match: TgxYWFmND-1NTYwNTgzMDM3N
+        - text: >
+            {
+              "client_id": "514233951460-ptar9o2diyaiz06uxa9inr2iuqjlvg1u.apps.googleusercontent.com",
+              "client_secret": "TgxYWFmND-1NTYwNTgzMDM3N",
+              "refresh_token": "1/QzlJ1Z8CY_phper0QGxDqU8sZwy9ZH-hKs4ZSbke27k",
+              "type": "authorized_user"
+            }
+          containsSecret: true
+          match: TgxYWFmND-1NTYwNTgzMDM3N
+        - text: >
+              const char token_str[] =
+                "{ \"client_id\": \"514233951460.apps.googleusercontent.com\","
+                "  \"client_secret\": \"TgxYWFmND-1NTYwNTgzMDM3N\","
+                "  \"refresh_token\": \"1/QzlJ1Z8CY_phper0QGxDqU8sZwy9ZH-hKs4ZSbke27k\","
+                "  \"type\": \"authorized_user\"}";
+          containsSecret: true
+          match: TgxYWFmND-1NTYwNTgzMDM3N
+        - text: >
+            "google" {
+              "class" = "io.gearpump.services.security.oauth2.impl.GoogleOAuth2Authenticator"
+              "callback" = "http://127.0.0.1:8090/login/oauth2/google/callback"
+              "clientid" = "514233951460-ptar9o2diyaiz06uxa9inr2iuqjlvg1u.apps.googleusercontent.com"
+              "clientsecret" = "TgxYWFmND-1NTYwNTgzMDM3N"
+              "default-userrole" = "guest"
+              icon = "/icons/google.png"
+            }
+          containsSecret: true
+          match: TgxYWFmND-1NTYwNTgzMDM3N
+        - text: >
+            OAUTH_CLIENT_ID=AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA.apps.googleusercontent.com
+            OAUTH_CLIENT_SECRET=BBBBBBBBBBBBBBBBBBBBBBBB
+          containsSecret: false
+        - text: >
+            GOOGLE_CLIENT_ID='sample_514233951460-ptar9o2diyaiz06uxa9inr2iuqjlvg1u.apps.googleusercontent.com'
+            GOOGLE_CLIENT_SECRET='sample_ND-1NTYwNTgzMDM3N'
+          containsSecret: false
+
+    - id: google-oauth-secret-no-prefix-client-id-after
+      metadata:
+        name: Google OAuth client secrets should not be disclosed
+      detection:
+        matching:
+          # "secret"
+          # any text, containing up to 1 newline
+          # the target client secret
+          # any text, containing up to 3 newlines
+          # ".apps.googleusercontent.com"
+          pattern: "(?i)(?:secret(?:[^\\r\\n]*+\\r?\\n)?[^\\r\\n]+)\\b([A-Za-z0-9_\\-]{24})\\b(?:(?:[^\\r\\n]*+\\r?\\n){0,3}[^\\r\\n]*\\.apps\\.googleusercontent\\.com\\b)"
+        post:
+          patternNot: "(?i)sample|example|x{6}|0{6}"
+          statisticalFilter:
+            threshold: 4.0
```

Same thing here. 3.8 might be a more accurate threshold.
```suggestion
            threshold: 3.8
```